src/appMain/life_cycle.cc

```diff
@@ -88,6 +93,15 @@ LifeCycle::LifeCycle()
 
 bool LifeCycle::StartComponents() {
   LOG4CXX_AUTO_TRACE(logger_);
+
+#ifdef OS_WINDOWS
+  WSAData wsa_data;
+  if (0 != WSAStartup(MAKEWORD(2, 2), &wsa_data)) {
```

Comment:
I think this is good solution, but not in our case. we have a few places, where we use raw sockets. streamer, tcp transport and possibly somewhere else. So keeping win socket init& cleanup here will not affect other parts. Until we will be sure that all raw sockets are using Socket, better to use explicit WSAStartup& cleanup in the Start/Stop components